pkg/covermerger/bq_csv_reader.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2024 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package covermerger
+
+import (
+	"compress/gzip"
+	"context"
+	"fmt"
+	"io"
+	"regexp"
+
+	"cloud.google.com/go/bigquery"
+	"cloud.google.com/go/civil"
+	"github.com/google/syzkaller/pkg/gcs"
+	"github.com/google/uuid"
+)
+
+var allowedFilePath = regexp.MustCompile(`^[./_a-zA-Z0-9]*$`)
+
+func isAllowedFilePath(s string) bool {
+	return allowedFilePath.MatchString(s)
+}
+
+type bqCSVReader struct {
+	closers   []io.Closer
+	gcsFiles  []io.Reader
+	ErrorChan chan error
+}
+
+func (r *bqCSVReader) Close() {
+	for _, c := range r.closers {
+		c.Close()
+	}
+}
+
+func MakeBQCSVReader() *bqCSVReader {
+	return &bqCSVReader{
+		ErrorChan: make(chan error, 1),
+	}
+}
+
+func (r *bqCSVReader) InitNsRecords(ctx context.Context, ns, filePath string, from, to civil.Date) error {
+	if !isAllowedFilePath(filePath) {
```

Comment:
In the only place where you call this function, the `filePath` argument is empty. Do you really need it?

Reply:
Yes. I'm building a page with single file coverage visualization.
This function is used there to get a single file coverage.

Reply:
It blocks #5173 . Please +1.